### DISCUSSION TITLE
Labo 2 : GameOver

### DISCUSSION THREAD

**Discussion by dylanramos** [web-classroom/web-classroom.github.io#52] [Q&A : Pratique / Labos] [answered]
J'ai une question concernant la fin d'une partie. Que faut-il faire exactement?
- Faut-il clear les 2 map?
- Est-ce que la partie est sensée recommencer automatiquement?
- Faut-il appeler la fonction gameOver() depuis step() ou depuis addNewShape()?
- A quoi sert la variable isGameOver s'il suffit d'appeler la fonction?

Merci beaucoup pour vos réponses.

**Comment by Nortalle** [comment, marked as answer]
Bonjour @dylanramos,

- Oui, les deux map doivent être clear.
- La partie ne recommence pas automatiquement
- `addNewShape()` appelle la fonction `gameOver()`.
- La valeur de `isGameOver` est modifiée depuis  `gameOver()`. Et c'est dans `step()` que cette valeur est vérifiée afin d'arrêter oui ou non la partie.

**Reply by dylanramos** [reply]
Merci beaucoup pour votre réponse
